compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -882,7 +883,19 @@ fn should_encode_mir(tcx: TyCtxt<'_>, def_id: LocalDefId) -> (bool, bool) {
             let is_const_fn = tcx.is_const_fn_raw(def_id.to_def_id())
                 || tcx.has_attr(def_id.to_def_id(), sym::default_method_body_is_const);
             let always_encode_mir = tcx.sess.opts.debugging_opts.always_encode_mir;
-            (is_const_fn, needs_inline || always_encode_mir)
+
+            let is_collect_trait_fn = {
+                let col_did = match tcx.lang_items().require(LangItem::Collectable) {
```

Comment:
I think this can be `let cold_id = tcx.lang_items().require(...).unwrap_or_else(|e| tcx.sess.fatal(&e)` if you prefer?

Reply:
Yeah that looks better. I've replaced this in all the places it comes up. Try this: cc5d6a0424832ee7de7c52eca1ee90e6d755b57b